src/sql-parser/src/parser.rs

```diff
@@ -2170,7 +2170,7 @@ impl Parser {
         match self.next_token() {
             // Do not accept `AS OF`, which is reserved for providing timestamp information
             // to queries.
-            Some(Token::Word(ref w)) if w.value == "OF" => {
+            Some(Token::Word(ref w)) if w.value.eq_ignore_ascii_case("OF") => {
```

Comment:
I believe you actually want `if w.keyword == "OF"`, which centralizes handling of how to match keywords case-insensitively. (Weirdly `parse_keyword` calls `eq_ignore_ascii_case` but that looks superfluous to me.)